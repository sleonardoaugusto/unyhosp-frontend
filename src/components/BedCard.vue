<template>
  <div class="card col-md-3">
    <p class="card__name">{{ bed.name }}</p>
    <div v-if="hasPacient" class="card__preview">
      <p class="card__text-content">
        Paciente:
        <router-link :to="{name: 'attendance', params: {pacientId: bed.pacient.id}}">{{ bed.pacient.name }}</router-link>
      </p>
      <p class="card__text-content">
        Data:
        <span>{{ bed.date }}</span>
      </p>
      <p class="card__text-content">
        DiagnÃ³stico:
        <span>{{ bed.pacient.diagnostic }}</span>
      </p>
      <p class="card__text-content">
        Data:
        <span>{{ bed.pacient.health_plan }}</span>
      </p>
    </div>
    <div v-else>
      <p class="card__text-content">
        Vazio
      </p>
    </div>
  </div>
</template>

<script>
export default {
  name: 'BedCard',
  props: {
    bed: {
      type: Object,
      required: true,
      default: () => ({})
    }
  },
  computed: {
    hasPacient() {
      return this.bed.pacient
    }
  }
}
</script>

<style scoped>

</style>