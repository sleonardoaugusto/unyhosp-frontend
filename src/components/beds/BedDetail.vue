<template>
  <div>
    <h1>Bed Detail Page</h1>
  </div>
</template>

<script>
export default {
  name: 'BedDetail',
  props: ['bedId']
}
</script>

<style scoped>

</style>