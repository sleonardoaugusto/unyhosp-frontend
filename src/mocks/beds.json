{
  "data": [
    {
      "id": 1,
      "name": "Box A",
      "uti_id": 1,
      "date": "09/12/2019",
      "pacient": {
        "name": "Maria Joaquina",
        "diagnostic": "Apendicite",
        "health_plan": "UNIMED"
      }
    },
    {
      "id": 2,
      "name": "Box B",
      "uti_id": 1,
      "date": "02/12/2019",
      "pacient": {
        "name": "Pedro Alvares",
        "diagnostic": "Dengue",
        "health_plan": "Santa Casa"
      }
    },
    {
      "id": 3,
      "name": "Box C",
      "uti_id": 1,
      "date": null,
      "pacient": null
    }
  ]
}