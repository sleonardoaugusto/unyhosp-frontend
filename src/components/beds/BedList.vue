<template>
  <div class="col-md-12 d-flex justify-content-center flex-wrap">
    <bed-card v-for="(b, i) in beds" :bed="b" :key="i"/>
  </div>
</template>

<script>
import BedCard from '@/components/beds/BedCard'

export default {
  name: 'BedList',
  components: {
    BedCard
  },
  data: () => ({
    beds: []
  }),
  created() {
    /* TODO
    *   Get beds*/
  }
}
</script>

<style scoped>

</style>