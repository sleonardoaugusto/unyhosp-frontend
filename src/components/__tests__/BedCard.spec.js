import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueRouter from 'vue-router'
import BedCard from '@/components/BedCard'
import beds from '@/components/__tests__/__mocks__/beds'

const localVue = createLocalVue()
localVue.use(VueRouter)

describe('BedCard.vue', () => {
  let wrapper
  const propsData = { bed: beds.data[0] }

  beforeEach(() => {
    wrapper = shallowMount(BedCard, {
      localVue,
      propsData
    })
  })

  it('component is rendering properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})