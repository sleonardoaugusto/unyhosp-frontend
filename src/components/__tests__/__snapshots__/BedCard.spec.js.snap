// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BedCard.vue component is rendering properly 1`] = `
<div class="card col-md-3">
  <p class="card__name">Box A</p>
  <div class="card__preview">
    <p class="card__text-content">
      Paciente:
      <a>Maria Joaquina</a></p>
    <p class="card__text-content">
      Data:
      <span>09/12/2019</span></p>
    <p class="card__text-content">
      Diagn√≥stico:
      <span>Apendicite</span></p>
    <p class="card__text-content">
      Data:
      <span>UNIMED</span></p>
  </div>
</div>
`;
